src/CarbonAware.DataSources/CarbonAware.DataSources.WattTime/src/WattTimeDataSource.cs

```diff
@@ -72,12 +72,15 @@ public async Task<EmissionsForecast> GetCurrentCarbonIntensityForecastAsync(Loca
             BalancingAuthority balancingAuthority = await this.GetBalancingAuthority(location, activity);
             var data = await this.WattTimeClient.GetCurrentForecastAsync(balancingAuthority);
 
+            var duration = GetDurationFromGridEmissionDataPoints(data.ForecastData.FirstOrDefault(), data.ForecastData.Skip(1)?.FirstOrDefault());
+            
             // Linq statement to convert WattTime forecast data into EmissionsData for the CarbonAware SDK.
             var forecastData = data.ForecastData.Select(e => new EmissionsData() 
             { 
                 Location = e.BalancingAuthorityAbbreviation, 
```

Comment:
Interesting.  Yea, I think it's out of scope for this PR, but it could be helpful to have an abstraction for common unit conversions in this domain.  Typically I like to wait until there is another use-case to justify the abstraction and then pull it out at that time.